typeobjectpattern/src/main/java/com/iluwatar/typeobject/JsonParser.java

```diff
@@ -47,23 +43,21 @@ public class JsonParser {
     this.candies = new Hashtable<>();
   }
 
-  void parse() throws IOException, ParseException {
-    var parser = new JSONParser();
-    var workingDirectory = new File("").getAbsolutePath();
-    var filePath = List.of("src", "main", "java", "com", "iluwatar", "typeobject", "candy.json");
-    var absolutePath = workingDirectory + File.separator + String.join(File.separator, filePath);
-    var jo = (JSONObject) parser.parse(new FileReader(absolutePath));
-    var a = (JSONArray) jo.get("candies");
-    for (var o : a) {
-      var candy = (JSONObject) o;
-      var name = (String) candy.get("name");
-      var parentName = (String) candy.get("parent");
-      var t = (String) candy.get("type");
+  void parse() throws JsonParseException {
+    var is = this.getClass().getClassLoader().getResourceAsStream("candy.json");
+    var reader = new InputStreamReader(is);
+    var json = (JsonObject) com.google.gson.JsonParser.parseReader(reader);
```

Comment:
Need to import this package com.google.gson.JsonParser.parseReader(reader)and can be used 
 JsonParser.parseReader(reader);

Reply:
Works only if I rename the class `JsonParser`.

Reply:
is there any other solution to fix it